File: sample_repo/repository/user_repository.py
"""Persistence boundary for user records."""

from repository.db import execute, query_by


class UserRepository:
    """Stores and retrieves user rows."""

    table = "users"

    def persist(self, user: dict) -> dict:
        """Write a user row to the database."""
        return execute(self.table, user)

    def find_by_email(self, email: str) -> dict:
        """Look a user up by email address."""
        return query_by(self.table, "email", email)


def get_repository() -> UserRepository:
    """Return the repository instance used by the services layer."""
    return UserRepository()

File: sample_repo_ts/src/repository/userRepository.ts
import { execute, queryBy } from "./db";

export class UserRepository {
  table = "users";

  persist(user: Record<string, unknown>) {
    return execute(this.table, user);
  }

  findByEmail(email: string) {
    return queryBy(this.table, "email", email);
  }
}

export function getRepository(): UserRepository {
  return new UserRepository();
}
